### PR TITLE
Make our CI policy clearer and more explicit

### PR DESCRIPTION
**Pull request by Zimmi48**

### CODE REVIEW

File: dev/ci/README.md [hunk 1]
```diff
@@ -0,0 +1,118 @@
+Continuous Integration
+======================
```
Comment (ejgallego) [inline dev/ci/README.md]: Maybe you'd like to add "for the Coq Proof Assistant" as to make the context clearer.

File: dev/ci/README.md [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+
+
+Information for external library / plugin authors
+----------------------------------------------------
+
+Changes to Coq are systematically tested for regression and compatibility
+breakage on our Continuous Integration (CI) platforms *before* integration.
+These tests include the compilation of several external libraries / plugins.
+
```
Comment (ejgallego) [inline dev/ci/README.md]: Maybe the intro could be moved before the first section. You could outline the global goals of the CI system, then mention that the document contains information for both "external" developers and "internal" ones.

File: dev/ci/README.md [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+  If this is not the case, consider adding a "lite" target that compiles just
+  part of it.
+
+In case you forget to comply with these last three conditions, we would reach
+out to you and give you a 30-day grace period during which your development
+would be moved into our "allow failure" category. At the end of the grace
+period, in the absence of progress, the development would be removed from our
+CI.
+
+**Add your development by submitting a pull request:** add a new `ci-mydev.sh`
```
Comment (ejgallego) [inline dev/ci/README.md]: Why is this not a markdown heading?

**Review by ejgallego — APPROVED** [review]
Looks good to me! We should certainly refine it more. I suggest we try to get somebody non-familiar with the document to read it and give us feedback. Maybe @aa755 , @beta-ziliani ?

**Comment by ejgallego** [comment]
Another minor touch to the document is IMHO to specify that the policy is not a hard policy "set in stone".

We currently accommodate CoLoR (thou I should get in touch with Frederic so a git migration happens) and SF, which is a special case, but important enough to deserve the support.

**Comment by beta-ziliani** [comment]
It's clear and to the point. Appreciated!

**Comment by Zimmi48** [comment]
@ejgallego: Thanks for your comments, now taken into account!
@beta-ziliani: Thanks for the quick feedback!

**Comment by aa755** [comment]
Sorry for the late response. Thanks for this initiative; it would be very useful for plugin/library authors.

As a minor comment, I think more details can be added to the steps in "Add your development by submitting a pull request". (Maybe this is because I have never used CI services.)
For example, what should the script `ci-mydev.sh` do (and not do)? Looking at the examples in `dev/ci/', I guess this script does the testing and success is indicated by the return code of the script?
If this is common knowledge, perhaps include a link to a suitable tutorial/reference?
Also, consider adding links to some pull requests that implement these steps for some libraries?

**Comment by Zimmi48** [comment]
@aa755 Thank you for your review (less than 24-hour delay does not count as "late response").

I agree that this paragraph was a bit scarce. I was afraid I'd be too verbose. No, this is not common CI knowledge: it is all very specific to Coq CI, and still experimental. I added links to two `ci-...` scripts that are among the simplest to give readers an idea of what is expected. Yes, the success is indicated by the return code, is it clearer with these examples? For now, we have no PR implementing these steps all at once correctly, but as soon as we get one, I'll make sure to add the link.

**Comment by maximedenes** [comment]
@Zimmi48 Can I merge this PR?

**Comment by Zimmi48** [comment]
@maximedenes Yes ;-)

**Pull request merged**